src/main/java/com/aws/greengrass/shadowmanager/sync/CloudDataClient.java

```diff
@@ -48,9 +49,9 @@ public class CloudDataClient {
     private final MqttClient mqttClient;
     private final ExecutorService executorService;
     @Getter(AccessLevel.PACKAGE)
-    private final Set<String> subscribedUpdateShadowTopics = new HashSet<>();
+    private final Set<String> subscribedUpdateShadowTopics = ConcurrentHashMap.newKeySet();
```

Comment:
don't need this anymore since `subscribedUpdateShadowTopics` and `subscribedDeleteShadowTopics` are now fully protected by `subscriptionLock` now

Reply:
Yes, with all the current changes we got, but shouldn't hurt to keep it that way.